packages/shared/routes/onboarding/views/network-setup/views/ChooseProtocolView.svelte

```diff
@@ -1,38 +1,38 @@
 <script lang="typescript">
+    import { onMount } from 'svelte'
     import { Animation, OnboardingButton, OnboardingLayout, Text } from 'shared/components'
+    import features from '@features/features'
     import {
         cleanupOnboarding,
-        createNewProfile,
-        initProfileManagerFromNewProfile,
-        newProfile,
-        updateNewProfile,
+        initialiseOnboardingProfile,
+        onboardingProfile,
+        shouldUseDeveloperProfile,
+        updateOnboardingProfile,
     } from '@contexts/onboarding'
-    import { AppStage, appStage, mobile } from '@core/app'
+    import { mobile } from '@core/app'
     import { localize } from '@core/i18n'
     import { NetworkProtocol, NetworkType } from '@core/network'
-    import { appRouter } from '@core/router'
-    import features from 'shared/features/features'
-    import { onMount } from 'svelte'
+    import { networkSetupRouter } from '@core/router'
 
-    async function onClick(networkProtocol: NetworkProtocol): Promise<void> {
-        if ($newProfile?.isDeveloperProfile) {
-            updateNewProfile({ networkProtocol })
+    function onProtocolSelectionClick(networkProtocol: NetworkProtocol): void {
+        if ($onboardingProfile?.isDeveloperProfile) {
+            updateOnboardingProfile({ networkProtocol })
         } else {
-            updateNewProfile({ networkProtocol, networkType: NetworkType.Mainnet })
-            await initProfileManagerFromNewProfile()
+            updateOnboardingProfile({ networkProtocol, networkType: NetworkType.Mainnet })
         }
-        $appRouter.next()
+        $networkSetupRouter.next()
     }
 
     async function onBackClick(): Promise<void> {
         await cleanupOnboarding(true)
-        $appRouter.previous()
+        $networkSetupRouter.previous()
     }
 
     onMount(() => {
-        if (!$newProfile?.id) {
-            createNewProfile({ isDeveloperProfile: $appStage !== AppStage.PROD })
+        if (!$onboardingProfile?.id) {
+            initialiseOnboardingProfile(shouldUseDeveloperProfile())
```

Comment:
shouldn't be hardcoded to use a developer profiule if not on mainnet, the user can be in this flow with developer or none developer profiles irrespective of the stage of the packaged app. Why do we need to initialise the onboarding profile at this point why has it gone?